experiments/pipeline/src/main/scala/org/apache/gearpump/experiments/pipeline/Processors.scala

```diff
@@ -0,0 +1,362 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.experiments.pipeline
+
+import akka.actor.actorRef2Scala
+import org.apache.gearpump.cluster.UserConfig
+import org.apache.gearpump.experiments.pipeline.Messages.Handlers.MemoryHandler
+import org.apache.gearpump.experiments.pipeline.Messages.{Body, Cpu, Envelope, Memory}
+import org.apache.gearpump.streaming.kafka.KafkaSource
+import org.apache.gearpump.streaming.kafka.lib.KafkaConfig
+import org.apache.gearpump.streaming.task.{StartTime, Task, TaskContext}
+import org.apache.gearpump.streaming.transaction.api.{MessageDecoder, TimeReplayableSource, TimeStampFilter}
+import org.apache.gearpump.{Message, TimeStamp}
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.hbase.client._
+import org.apache.hadoop.hbase.util.Bytes
+import org.apache.hadoop.hbase.{HColumnDescriptor, HTableDescriptor, TableName}
+import upickle._
+
+import scala.collection.JavaConversions._
+import scala.collection.immutable.HashMap
+
+
+// NOTE: Do not split up into separate files
+// See http://stackoverflow.com/questions/28630780/upickle-and-scalajs-sealed-trait-serialisation
+object Messages {
+
+  sealed trait Datum[T] {
+    val dimension: String
+    val metric: String
+    val value: T
+  }
+
+  case class Cpu(dimension: String, metric: String, value: Double) extends Datum[Double]
+
+  case class Memory(dimension: String, metric: String, value: Long) extends Datum[Long]
+
+  case class Body(source_id: String, event_ts: String, metrics: Array[Datum[_]])
+
+  case class Envelope(id: Long, on: String, body: String)
+
+  trait Handler[T] {
+    def handle(data: String): Datum[T]
+  }
+
+  object Handlers {
+    implicit object CpuHandler extends Handler[Double] {
+      override def handle(data: String): Cpu = {
+        read[Cpu](data)
+      }
+    }
+    implicit object MemoryHandler extends Handler[Long] {
+      override def handle(data: String): Memory = {
+        read[Memory](data)
+      }
+    }
+  }
+  object Converters {
+    def convert[T](data: String)(implicit handler: Handler[T]): Datum[T] = {
+      handler.handle(data)
+    }
+  }
+
+}
+
+object Base64 {
+  private[this] val zero = Array(0, 0).map(_.toByte)
```

Comment:
I think you can use guava to do the base64 encoding stuff.

http://stackoverflow.com/a/18241113